src/EditorFeatures/CSharpTest/MisplacedUsingDirectives/MisplacedUsingDirectivesInCompilationUnitCodeFixProviderTests.cs

```diff
@@ -0,0 +1,418 @@
+
+            var fixedTestCode = @"// File Header
+
+namespace TestNamespace
+{
+    // Leading Comment
+
+    {|Warning:using Microsoft.CodeAnalysis;|}
+    {|Warning:using System;|}
+}
+";
+
+            return TestInRegularAndScriptAsync(testCode, fixedTestCode, InsideNamespaceOption, placeSystemNamespaceFirst: true);
+        }
+
+        /// <summary>
+        /// Verifies that a code fix will not be offered for MisplacedUsing diagnostics when multiple namespaces are present.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithMultipleNamespaces_NoCodeFixOffered()
+        {
+            var testCode = @"[|using System;|]
+
+namespace TestNamespace1
+{
+    public class TestClass1
+    {
+    }
+}
+
+namespace TestNamespace2
+{
+}
+";
+
+            return TestMissingAsync(testCode, InsideNamespaceOption);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will properly move pragmas.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithPragma_PragmaMoved()
+        {
+            var testCode = @"#pragma warning disable 1573 // Comment
+[|using System;
+using System.Threading;|]
+
+namespace TestNamespace
+{
+}
+";
+
+            var fixedTestCode = @"namespace TestNamespace
+{
+#pragma warning disable 1573 // Comment
+    {|Warning:using System;|}
+    {|Warning:using System.Threading;|}
+}
+";
+
+            return TestInRegularAndScriptAsync(testCode, fixedTestCode, InsideNamespaceOption, placeSystemNamespaceFirst: true);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will properly move regions.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithRegion_RegionMoved()
+        {
+            var testCode = @"#region Comment
+#endregion Comment
+[|using System;
+using System.Threading;|]
+
+namespace TestNamespace
+{
+}
+";
+
+            var fixedTestCode = @"namespace TestNamespace
+{
+    #region Comment
+    #endregion Comment
+    {|Warning:using System;|}
+    {|Warning:using System.Threading;|}
+}
+";
+
+            return TestInRegularAndScriptAsync(testCode, fixedTestCode, InsideNamespaceOption, placeSystemNamespaceFirst: true);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will properly move comment trivia.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithCommentTrivia_TriviaMoved()
+        {
+            var testCode = @"
+// Some comment
+[|using System;
+using System.Threading;|]
+
+namespace TestNamespace
+{
+}
+";
+
+            var fixedTestCode = @"namespace TestNamespace
+{
+
+    // Some comment
+    {|Warning:using System;|}
+    {|Warning:using System.Threading;|}
+}
+";
+
+            return TestInRegularAndScriptAsync(testCode, fixedTestCode, InsideNamespaceOption, placeSystemNamespaceFirst: true);
+        }
+
```

Comment:
Do we have a unit test when original code has some using inside namespace and some outside namespace?